test/tests/rackhd20/test_rackhd20_api_ucs.py

```diff
@@ -0,0 +1,297 @@
+                logs.info_1("Deleting OBM: {0}. Status was: {1}".format(obm['id'], str(api_data['status'])))
+                time.sleep(self.MAX_WAIT_ON_DELETE)
+        api_data = fit_common.rackhdapi('/api/2.0/obms')
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 200, got:' + str(api_data['status']))
+        temp = {}
+        for obm in api_data['json']:
+            temp[obm['id']] = obm['service']
+        logs.info_1("Found {0} ucs obms remaining after restoring the obms, should be {1}. Remaining OBMs: {2}"
+                    .format(len(temp), len(self.INITIAL_OBMS), temp))
+
+    def ucs_url_factory(self, api, identifier=None):
+        """
+        returns a fully qualified UCS API
+        :param api:UCS API
+        :param identifier: identify the ucs element in the catalog API
+        :return:
+        """
+        ucs_service = self.RACKHD_IP + ":" + self.UCS_PORT + "/"
+        ucs_manager = self.UCS_IP
+        if identifier is None:
+            url = "http://" + ucs_service + api + "?host=" + ucs_manager + " &user=ucspe&password=ucspe"
+        else:
+            url = "http://" + ucs_service + api + "?host=" + ucs_manager + " &user=ucspe&password=ucspe"\
+                  "&identifier=" + identifier
+        return url
+
+    def test_api_20_ucs_log_in(self):
+        """
+        Test the /logIn ucs API
+        :return:
+        """
+        url = self.ucs_url_factory("login")
+        api_data = fit_common.restful(url)
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 201, got:' + str(api_data['status']))
+
+        self.assertNotEqual(api_data["json"], None, "Expected a token to be returned on login and received None")
+        self.assertNotEqual(type(api_data["json"]), "unicode", "Unexpected Token was received on Login")
+
+    def test_api_20_ucs_get_sys(self):
+        """
+        Test the /sys ucs API
+        :return:
+        """
+        url = self.ucs_url_factory("sys")
+        api_data = fit_common.restful(url)
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 201, got:' + str(api_data['status']))
+
+        self.assertEqual(len(api_data["json"]["Fabric Interconnects"]), 2,
+                         "Expected 2 Fabric Interconnects but found {0}"
+                         .format(len(api_data["json"]["Fabric Interconnects"])))
+        self.assertEqual(len(api_data["json"]["Servers"]), 7, "Expected 7 Servers but found {0}"
+                         .format(len(api_data["json"]["Servers"])))
+        self.assertEqual(len(api_data["json"]["FEX"]), 2, "Expected 2 FEX but found {0}"
+                         .format(len(api_data["json"]["FEX"])))
+        self.assertEqual(len(api_data["json"]["Chassis"]), 4, "Expected 2 Chassis but found {0}"
+                         .format(len(api_data["json"]["Chassis"])))
```

Comment:
design test question - do you want the test to check each of these items or exit on the first error hit?
if check each item and report errors, consider adding the errors to an errorlist, and then assert at the end if errorlist is not empty (examples of this in a couple scripts, ie. ./compute/test_rackhd20_computenode_pollers.py)